Frameworks/include/CoreTextInternal.h

```diff
@@ -32,38 +36,48 @@
 
 @interface _CTFrameSetter : NSObject {
 @public
-    idretaintype(_CTTypesetter) _typesetter;
+    StrongId<_CTTypesetter> _typesetter;
 }
 @end
 
 @interface _CTFrame : NSObject {
 @public
     _CTFrameSetter* _frameSetter;
     CGRect _frameRect;
+    StrongId<NSMutableArray> _lines;
+
+    // TODO::
+    // Do we need these anymore?
     CGSize _totalSize;
-    idretaintype(NSMutableArray) _lines;
     std::vector<CGPoint> _lineOrigins;
 }
 @end
 
 @interface _CTLine : NSObject {
 @public
-    NSRange _strRange;
+    _CTTypesetter* _typeSetter;
+    CFRange _strRange;
+    CGPoint _lineOrigin;
```

Comment:
>_lineOrigin [](start = 12, length = 11)

Will DWrite give origins for the line or for each glyphrun? If the latter we should keep this variable in the runs